### DISCUSSION TITLE
Set session variable from JS

### DISCUSSION THREAD

**Discussion by notflip** [statamic/cms#7045] [Q&A]
I'm using AlpineJS to have a quiz on my page, when someone completes the quiz I would like to set the score in the session, so I can send it along a form deeper down the page.

I can use `{{ session:set score="10" }}` to set the score, but when I add this to a function in JS, it is executed immediately, the code is parsed by antlers even if the JS function is not ran.

**Does anyone know if I can set Statamic session data from within JS?**

**Comment by duncanmcclean** [comment]
Antlers is parsed **before** the browser receives the HTML for the page so any code you have there won't be dependent what happens within your JavaScript.

**Comment by jasonvarga** [comment]
You cannot set session data from javascript.

If you want to set session data from js, you'll need to do an ajax request to somewhere on the server and set it within php.

You can set cookies from JS, though.
